src/utils.ts

```diff
@@ -0,0 +1,23 @@
+const cardanoCrypto = require('cardano-crypto.js');
+import {
+  ADA_DERIVATION_PATH,
+  HARDENED_THRESHOLD,
+} from './constants';
+/**
+ * Generate Ada key (private key or public key)
+ * @param mnemonic mnemonic seed to use
+ * @param startInd An integer that specifies where to start the selection in the slice() function
+ * @param endInd An integer that specifies where to end the selection in the slice() function
+ * @returns string
+ */
+export const generateAdaKey = async (mnemonic: string, privateKey: boolean): Promise<string> => {
```

Comment:
`privateKey` is a very poor name for a `boolean`. `isPrivateKey`, `bPrivateKey` or `genPrivateKey` could have been way better.